src/Extension/Evaluator.cs

```diff
@@ -23,5 +27,35 @@ public Evaluation EvaluateResult(KernelCommandResult result)
             }
             return new Evaluation { Passed = true };
         }
+
+        public void AddInputExecutionCriterion(int cellId, string code)
+        {
+            executionCriteria.Remove(cellId);
+            executionCriteria.Add(cellId, code);
+        }
+
+        public string GetInputExecutionCriterion(int cellId)
```

Comment:
Is the returned value a `string`? Can we come up with a more semantically distinct type?

Reply:
Created criterion types. Currently, the classes are minimal but we might expect more functionality to be added.